### PR TITLE
feat: receive response from listeners

### PR DESCRIPTION
**Pull request by pichoemr**
linked to https://github.com/lenra-io/application-runner/issues/247

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [ ] I included unit tests that cover my changes
- [ ] I added/updated the documentation about my changes
- [ ] I made my own code-review before requesting one

## Description of the changes

remove notifier and call a callback when push receive response

### CODE REVIEW

**Review by jonas-martinez — APPROVED** [review]

**Review by jonas-martinez — CHANGES_REQUESTED** [review]
Most of the tests are broken

I can take care of it if you want

**Review by taorepoara — APPROVED** [review]
Seems OK

**Comment by taorepoara** [comment]
You should fix the showcase example

File: lib/models/channel_model.dart [hunk 1]
```diff
@@ -41,9 +41,8 @@ class ChannelModel extends ChangeNotifier {
     return this;
   }
 
-  bool handleNotifications(Event notification) {
-    channel!.send('run', notification.toMap());
-    return true;
+  void sendEvent(Event notification, Function(Map<dynamic, dynamic>) callback) {
```
Comment (Nesqwik) [inline lib/models/channel_model.dart]: could you change this function to return a future rather than a callback ? 
It is probably more convenient to use

Comment (Nesqwik) [reply lib/models/channel_model.dart]: Note: Use [Completer](https://api.dart.dev/stable/2.10.5/dart-async/Completer-class.html) to create a new future and add the receive("error") to handle the error response

**Review by Nesqwik — CHANGES_REQUESTED** [review]

File: lib/models/channel_model.dart [hunk 2]
```diff
@@ -41,9 +43,10 @@ class ChannelModel extends ChangeNotifier {
     return this;
   }
 
-  bool handleNotifications(Event notification) {
-    channel!.send('run', notification.toMap());
-    return true;
+  Future sendEvent(Event notification) {
+    final completer = Completer();
+    channel!.send('run', notification.toMap())!.receive("ok", (response) => completer.complete(response));
```
Comment (Nesqwik) [inline lib/models/channel_model.dart]: the `.receive("error", (error) => completer.completeError(error))` is missing to handle the error return

**Review by Nesqwik — CHANGES_REQUESTED** [review]

**Review by jonas-martinez — CHANGES_REQUESTED** [review]
This is some really good job ! Everything looks good to me.

Can you just fix the CI and add the proposed changes in nesqwik review ?

File: lib/models/channel_model.dart [hunk 3]
```diff
@@ -41,9 +43,13 @@ class ChannelModel extends ChangeNotifier {
...
 
-  bool handleNotifications(Event notification) {
-    channel!.send('run', notification.toMap());
-    return true;
+  Future sendEvent(Event notification) {
+    final completer = Completer();
+    channel!
+        .send('run', notification.toMap())!
+        .receive("ok", (body) => completer.complete(body))
+        .receive("error", (error) => completer.complete(error));
```
Comment (Nesqwik) [inline lib/models/channel_model.dart]: For the error, the completer should be completeError if i'm not mistaken
see : https://api.dart.dev/stable/2.10.5/dart-async/Completer/completeError.html

```suggestion
        .receive("error", (error) => completer.completeError(error));
```

**Review by Nesqwik — CHANGES_REQUESTED** [review]
One small change, fix the CI and we can merge this one :)

**Review by jonas-martinez — APPROVED** [review]

**Pull request merged**